LibreNMS/Authentication/RadiusAuthorizer.php

```diff
@@ -33,7 +33,47 @@ public function authenticate($credentials)
 
         $password = $credentials['password'] ?? null;
         if ($this->radius->accessRequest($credentials['username'], $password) === true) {
-            $this->addUser($credentials['username'], $password, Config::get('radius.default_level', 1));
+
+                //attribute 11 is "Filter-Id"
+            //Always set password change to 0 - password resides in AAA, not LibreNMS
+            //If attribute 11 is sent in reply after accept - update user
+            //If user exists - update, not add.
+            //If new user - add user with attribute value if present, or use default from config.
+
+            $attribute = '';
+            if ($this->radius->getAttribute(11)) {
+                switch ($this->radius->getAttribute(11)) {
+                        case 'lnms_admin':
```

Comment:
Rename these attributes to something like this, you could even use a match statement.
I renamed $attribute to $level  to make it easier to understand what the variable is and included the default level to save the if ($attribute) statements below.

```php
$level = match($this->radius->getAttribute(11)) {
    'librenms_role_admin' => 10,
    'librenms_role_normal' => 1,
    'librenms_role_global-read' => 5,
    default => Config::get('radius.default_level', 1)
};

if ($this->userExists($credentials['username'])) {
    // existing user, update the level only
    $user = \App\Models\User::find($this->getUserid($credentials['username']));
    $user->level = $level
    $user->save();
} else {
    $this->addUser($credentials['username'], $password, $level, '', $credentials['username'], 0, '');
}
```

Reply:
A couple of edits there after thinking about the code a bit.

Reply:
@kimhaak yes, there is no way to make the code support custom roles until LibreNMS does.

I just want the user configuration to be compatible so we only need to change the code to support RBAC.

Reply:
@kimhaak yes, I use the return that way for readability.  else is hard to read (maybe not one, but the more you have the harder it gets to follow).

https://levelup.gitconnected.com/how-you-can-avoid-using-else-in-your-code-871197a1adbc

Reply:
Fixed up my hand written code so you an commit it and test below.